TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/tests/test_replay.py

```diff
@@ -136,3 +105,73 @@ def test_nonexistent_replayer_type():
+
+
+@pytest.mark.parametrize(
+    "replayer_fixture, mock_desired_count, validate_desired_count",
+    [
+        ('ecs_replayer', _mock_ecs_set_desired_count, _mock_ecs_validate_desired_count),
+        ('k8s_replayer', _mock_k8s_set_desired_count, _mock_k8s_validate_desired_count),
+    ]
+)
+def test_replayer_start_sets_desired_count(request,
+                                           replayer_fixture,
+                                           mock_desired_count,
+                                           validate_desired_count,
+                                           mocker):
+    replayer = request.getfixturevalue(replayer_fixture)
+    mock = mock_desired_count(mocker)
+    replayer.start()
+    validate_desired_count(mock, replayer, replayer.config["scale"])
+
+
+@pytest.mark.parametrize(
+    "replayer_fixture, mock_desired_count, validate_desired_count",
+    [
+        ('ecs_replayer', _mock_ecs_set_desired_count, _mock_ecs_validate_desired_count),
+        ('k8s_replayer', _mock_k8s_set_desired_count, _mock_k8s_validate_desired_count),
+    ]
+)
+def test_replayer_stop_sets_desired_count(request,
+                                          replayer_fixture,
+                                          mock_desired_count,
+                                          validate_desired_count,
+                                          mocker):
+    replayer = request.getfixturevalue(replayer_fixture)
+    mock = mock_desired_count(mocker)
+    replayer.stop()
+    validate_desired_count(mock, replayer, 0)
+
+
+@pytest.mark.parametrize(
+    "replayer_fixture, mock_desired_count, validate_desired_count",
+    [
+        ('ecs_replayer', _mock_ecs_set_desired_count, _mock_ecs_validate_desired_count),
+        ('k8s_replayer', _mock_k8s_set_desired_count, _mock_k8s_validate_desired_count),
+    ]
+)
+def test_replayer_scale_sets_desired_count(request,
+                                           replayer_fixture,
+                                           mock_desired_count,
+                                           validate_desired_count,
+                                           mocker):
+    replayer = request.getfixturevalue(replayer_fixture)
+    mock = mock_desired_count(mocker)
+    replayer.scale(5)
+    validate_desired_count(mock, replayer, 5)
+
+
+@pytest.mark.parametrize("replayer_fixture", ['ecs_replayer', 'k8s_replayer'])
+def test_replayer_describe_no_json(request, replayer_fixture):
+    replayer = request.getfixturevalue(replayer_fixture)
+    success, output = replay_.describe(replayer, as_json=False)
+    assert success
+    assert output == yaml.safe_dump(replayer.config)
+
+
+@pytest.mark.parametrize("replayer_fixture", ['ecs_replayer', 'k8s_replayer'])
+def test_replayer_describe_as_json(request, replayer_fixture):
+    replayer = request.getfixturevalue(replayer_fixture)
+    success, output = replay_.describe(replayer, as_json=True)
+    assert success
+    assert json.loads(output) == replayer.config
```

Comment:
These parameterized tests are a nice improvement!

Reply:
That is my hope 🤞, and that future tests are easier to add. If we revisit later and say that I've abstracted these tests too much I'm happy to rethink them